src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+        }
+
+        return new InternationalFixedDate(prolepticYear, month, dayOfMonth);
+    }
+
+    /**
+     * Factory method, validates the given triplet year, month and dayOfMonth
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param isLeapDay     is it leap-day, month-less day between end of June and beginning of Sol
+     * @param isYearDay     is it year-day, month-less day between end of December of the old year and beginning of January of the following year
+     * @return the International fixed date
+     * @throws DateTimeException if the date is invalid
+     */
+    static InternationalFixedDate create(final int prolepticYear, final boolean isLeapDay, final boolean isYearDay) {
+        InternationalFixedChronology.YEAR_RANGE.checkValidValue(prolepticYear, ChronoField.YEAR_OF_ERA);
+
+        if (1 > prolepticYear) {
+            throw new DateTimeException("Invalid date: " + prolepticYear);
+        }
+
+        if (isLeapDay == isYearDay) {
+            throw new DateTimeException("Invalid date: " + prolepticYear + " both leap-day and year-day are " + isYearDay);
+        }
+
+        if (isLeapDay && !InternationalFixedChronology.INSTANCE.isLeapYear(prolepticYear)) {
+            throw new DateTimeException("Invalid leap day for year: " + prolepticYear);
+        }
+
+        return new InternationalFixedDate(prolepticYear, isLeapDay, isYearDay);
+    }
+
+    /**
+     *
+     * Validates the object.
+     *
+     * @return the resolved date, not null
+     */
+    private Object readResolve() {
+        if (isLeapDay() || isYearDay()) {
+            return InternationalFixedDate.of(prolepticYear, isLeapDay(), isYearDay());
+        }
+
+        return InternationalFixedDate.of(prolepticYear, getMonth(), getDayOfMonth());
+    }
+
+    //-----------------------------------------------------------------------
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int getProlepticYear() {
+        return prolepticYear;
+    }
+
+    public boolean isLeapDay() {
+        return isLeapYear() && dayOfYear == 169;
+    }
+
+    public boolean isYearDay() {
+        return dayOfYear == InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear() ? 1 : 0);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int getMonth() {
+        if (isYearDay() || isLeapDay()) {
+            return 0;
+        }
+
+        int doy = getDayOfYearAdjusted() - 1;
+
+        return 1 + (doy / InternationalFixedChronology.DAYS_IN_MONTH);
+    }
+
+    /**
+     * For calculation purposes, consider leap day to be associated with month Sol.
+     * In the same spirit, associate year day with the last month of the year.
+     * @return
+     */
+    private int getCalculatedMonth() {
+        return isYearDay() ?
+            InternationalFixedChronology.MONTHS_IN_YEAR : isLeapDay() ? 7 : getMonth();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    int getDayOfMonth() {
```

Comment:
Same as for `month()`.  I recommend `-1` for leap-day and `0` for year-day again.